### PR TITLE
Updated invalid_id_list() to be a general data function.

### LINKED ISSUE #3003: fix invalid_id_list() to be a general data function.
Remove 'id' attribute which is being passed from the function.

### PR DESCRIPTION
**Pull request by sthirugn**
Fixed #3003

### CODE REVIEW

**Comment by sthirugn** [comment]
Test result:
```
# nosetests tests/foreman/cli/test_architecture.py:TestArchitecture.test_negative_delete
.
----------------------------------------------------------------------
Ran 1 test in 4.199s

OK
# nosetests tests/foreman/cli/test_domain.py:TestDomain.test_negative_delete
.
----------------------------------------------------------------------
Ran 1 test in 4.443s

OK
# nosetests tests/foreman/cli/test_environment.py:TestEnvironment.test_negative_delete
.
----------------------------------------------------------------------
Ran 1 test in 4.531s

OK
# nosetests tests/foreman/cli/test_hostgroup.py:TestHostGroup.test_negative_delete
.
----------------------------------------------------------------------
Ran 1 test in 5.163s

OK
# nosetests tests/foreman/cli/test_model.py:TestModel.test_negative_delete
.
----------------------------------------------------------------------
Ran 1 test in 4.844s

OK
# nosetests tests/robottelo/test_datafactory.py
....
----------------------------------------------------------------------
Ran 4 tests in 0.022s

OK
```

**Comment by sthirugn** [comment]
Rebased

File: robottelo/datafactory.py [hunk 1]
```diff
@@ -64,10 +64,9 @@ def generate_strings_list(length=None, remove_str=None, bug_id=None):
 def invalid_id_list():
     """Generates a list of invalid IDs."""
     return [
-        {'id': gen_string('alpha')},
-        {'id': None},
```
Comment (elyezer) [inline robottelo/datafactory.py]: This will translate to `hammer organization delete`, so I think it should stay.

File: robottelo/datafactory.py [hunk 2]
```diff
@@ -64,10 +64,9 @@ def generate_strings_list(length=None, remove_str=None, bug_id=None):
...
-        {'id': ''},
-        {'id': -1},
+        gen_string('alpha'),
+        u'',
```
Comment (elyezer) [inline robottelo/datafactory.py]: This will translate to `hammer organization delete --id=`

File: robottelo/datafactory.py (hunk 1 above)
Comment (sthirugn) [inline robottelo/datafactory.py]: :+1:

**Comment by sthirugn** [comment]
Comments addressed and rebased

**Comment by omaciel** [comment]
ACK

**Comment by elyezer** [comment]
ACK

**Pull request merged**